app/objects/secondclass/c_fact.py

```diff
@@ -99,7 +99,7 @@ def __eq__(self, other):
             return self.unique == other.unique and self.source == other.source
         return False
 
-    def __init__(self, trait, value=None, score=1, source=None, origin_type=None, links=None,
+    def __init__(self, trait='', value=None, score=1, source=None, origin_type=None, links=None,
```

Comment:
This is potentially problematic, unfortunately. The spec for a Facts requires that a Fact always have a set Trait value. While whether or not this is necessarily the best situation is potentially up for debate, we kind of absolutely need it to be present (and set to something) in order to properly address/access Facts in the current core. Is there are specific problem you were trying to solve by giving it a default value here?

Reply:
I don't think this is absolutely necessary, but I was hoping to add some sort of request schema to this endpoint as well as the relationship endpoint. I thought it would be nice to have some information in the swagger docs about the expected request body, and it would allow for field validation as well.

When using both of these endpoints, are Fact.trait and Relationship.source expected to be in the request body? If so, I could just add a partial request schema - this would return an error if required fields are missing, or if field types are incorrect. If they aren't necessary, I might just remove validation all together, as I don't think there's a clean way to validate the received payloads in that case. Adding the default parameters in the Fact/Relationship objects was a way for me to validate in the latter case, but I can remove those